### PR TITLE
doc: release: add overview section

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Add back an overview section.

### CODE REVIEW

**Review by mmahadevan108 — DISMISSED** [review]

**Review by carlescufi — CHANGES_REQUESTED** [review]
This is not the overview, these are the new APIs? The overview typically is separate from this. I see that this is not properly done in the last few releases, can you please use this one as a template:

https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/release-notes-2.6.rst

**Comment by fabiobaltieri** [comment]
> This is not the overview, these are the new APIs? The overview typically is separate from this. I see that this is not properly done in the last few releases, can you please use this one as a template:
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/release-notes-2.6.rst

Oh, that's what you meant... I was wondering if I missed something and loocked back at the previous notes but I stopped at 2.7. :-) Oky I'll rework this in a bit.

**Comment by fabiobaltieri** [comment]
Ok how about now? Mentioned sysbuild, blobs, i3c, w1 and the fancy Kconfig thing that @galak did (was not mentioned anywhere else, I think it's cool, cleaned up a bunch of config and surfaced ton of misconfigs)

File: doc/releases/release-notes-3.2.rst [hunk 1]
```diff
@@ -447,6 +458,7 @@ Boards & SoC Support
   * MAX7219 LED display driver shield
   * Panasonic Grid-EYE (AMG88xx)
 
+
```
Comment (henrikbrixandersen) [inline doc/releases/release-notes-3.2.rst:461]: Nit

**Review by henrikbrixandersen — DISMISSED** [review]
Looks good.

**Comment by henrikbrixandersen** [comment]
Should we mention the deprecation of pinmux here?

File: doc/releases/release-notes-3.2.rst [hunk 2]
```diff
@@ -5,6 +5,17 @@
 Zephyr 3.2.0 (Working Draft)
 ############################
 
+We are pleased to announce the release of Zephyr version 3.2.0.
+
+Major enhancements with this release include:
```
Comment (stephanosio) [inline doc/releases/release-notes-3.2.rst:10]: I think this deserves a mention:
https://github.com/zephyrproject-rtos/zephyr/blame/main/doc/releases/release-notes-3.2.rst#L1333-L1335

Comment (fabiobaltieri) [reply doc/releases/release-notes-3.2.rst:10]: Oh yeah, I'll add it.

**Comment by fabiobaltieri** [comment]
> Should we mention the deprecation of pinmux here?

Yeah, maybe we can frame it as "Converted all supported boards from pinmux to pinctrl" instead?

**Comment by henrikbrixandersen** [comment]
> Yeah, maybe we can frame it as "Converted all supported boards from pinmux to pinctrl" instead?

Sounds good, yes.

**Comment by fabiobaltieri** [comment]
Added a reference to picolibc and pinctrl, dropped the stray whitespace.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Comment by mmahadevan108** [comment]
@carlescufi, can you please take a look again.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**